Tools/LambdaTestTool-v2/tests/Amazon.Lambda.TestTool.IntegrationTests/ApiGatewayEmulatorProcessTests.cs

```diff
@@ -290,15 +290,107 @@ public async Task TestLambdaWithNullEndpoint()
+            await StartTestToolProcessAsync(mode, "largerequestfunction", lambdaPort, apiGatewayPort, _cancellationTokenSource);
+            await WaitForGatewayHealthCheck(apiGatewayPort);
+
+            var handler = (APIGatewayProxyRequest request, ILambdaContext context) =>
+            {
+                var env = Environment.GetEnvironmentVariable("AWS_LAMBDA_RUNTIME_API");
+                testOutputHelper.WriteLine($"TestLambdaWithLargeRequestPayload {mode}: {env}");
+                return new APIGatewayProxyResponse
+                {
+                    StatusCode = 200,
+                    Body = request.Body.Length.ToString()
+                };
+            };
+
+            Environment.SetEnvironmentVariable("AWS_LAMBDA_RUNTIME_API", $"localhost:{lambdaPort}/largerequestfunction");
+            _ = LambdaBootstrapBuilder.Create(handler, new DefaultLambdaJsonSerializer())
+                .Build()
+                .RunAsync(_cancellationTokenSource.Token);
+
+            // Create a payload just over 6MB
+            var largePayload = new string('X', 6 * 1024 * 1024 + 1024); // 6MB + 1KB
+
+            var response = await TestEndpoint("largerequestfunction", apiGatewayPort, payload: largePayload);
+
+            Assert.Equal(HttpStatusCode.RequestEntityTooLarge, response.StatusCode);
+            var responseContent = await response.Content.ReadAsStringAsync();
+            Assert.Contains(mode == ApiGatewayEmulatorMode.Rest ? "Request Too Long" : "Request Entity Too Large", responseContent);
+        }
+        finally
+        {
+            await _cancellationTokenSource.CancelAsync();
+            Console.SetError(consoleError);
+        }
+    }
+
+    [Fact]
+    public async Task TestLambdaWithLargeRequestPayload_HttpV2()
+    {
+        var (lambdaPort, apiGatewayPort) = await GetFreePorts();
+        _cancellationTokenSource = new CancellationTokenSource();
+        _cancellationTokenSource.CancelAfter(TimeSpan.FromSeconds(120));
+        var consoleError = Console.Error;
+        try
+        {
+            Console.SetError(TextWriter.Null);
+            await StartTestToolProcessAsync(ApiGatewayEmulatorMode.HttpV2, "largerequestfunction", lambdaPort, apiGatewayPort, _cancellationTokenSource);
+            await WaitForGatewayHealthCheck(apiGatewayPort);
+
+            var handler = (APIGatewayHttpApiV2ProxyRequest request, ILambdaContext context) =>
+            {
+                var env = Environment.GetEnvironmentVariable("AWS_LAMBDA_RUNTIME_API");
+                testOutputHelper.WriteLine($"TestLambdaWithLargeRequestPayload HttpV2: {env}");
+                return new APIGatewayHttpApiV2ProxyResponse
+                {
+                    StatusCode = 200,
+                    Body = request.Body.Length.ToString()
+                };
+            };
+
+            Environment.SetEnvironmentVariable("AWS_LAMBDA_RUNTIME_API", $"localhost:{lambdaPort}/largerequestfunction");
```

Comment:
Coordinate with @philasmar PR so that you can set the API via a command line instead of environment variable. https://github.com/aws/aws-lambda-dotnet/pull/1992